src/commonMain/kotlin/org/cru/godtools/tool/model/Multiselect.kt

```diff
@@ -0,0 +1,99 @@
+package org.cru.godtools.tool.model
+
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import org.cru.godtools.tool.FEATURE_MULTISELECT
+import org.cru.godtools.tool.ParserConfig
+import org.cru.godtools.tool.internal.RestrictTo
+import org.cru.godtools.tool.internal.VisibleForTesting
+import org.cru.godtools.tool.state.State
+import org.cru.godtools.tool.xml.XmlPullParser
+import org.cru.godtools.tool.xml.parseChildren
+
+private const val XML_STATE = "state"
+private const val XML_SELECTION_LIMIT = "selection-limit"
+private const val XML_OPTION = "option"
+private const val XML_OPTION_VALUE = "value"
+
+class Multiselect : Content {
+    internal companion object {
+        internal const val XML_MULTISELECT = "multiselect"
+    }
+
+    @VisibleForTesting
+    internal val stateName: String
+    @VisibleForTesting
+    internal val selectionLimit: Int
+
+    val options: List<Option>
+
+    internal constructor(parent: Base, parser: XmlPullParser) : super(parent, parser) {
+        parser.require(XmlPullParser.START_TAG, XMLNS_CONTENT, XML_MULTISELECT)
+
+        stateName = parser.getAttributeValue(XML_STATE).orEmpty()
+        selectionLimit = (parser.getAttributeValue(XML_SELECTION_LIMIT)?.toIntOrNull() ?: 1).coerceAtLeast(1)
```

Comment:
So this just checks if the multi-select has a certain amount of selection options, and if not it defaults to 1?

Reply:
yes, a multiselect can have a defined selection limit which will limit the number of currently selected items.

This logic reads the string value from XML, converts it to an int, defaults to 1, and coerces whatever the value is to be at least 1